### ISSUE TITLE
hello，why prepocess all dataset sample frequency at 256hz？I find original TDBrain frequency is 500HZ

### ISSUE THREAD

**Issue by David-Ttao** [DL4mHealth/COMET#2]

**Comment by YiheWang** [comment]
This is an interesting question. The sampling frequency is often treated as a tuning parameter. Based on previous experiments, 256Hz has proven to be a good balance between memory efficiency and the minimization of information loss during downsampling.

If you have more GPUs available for training, you could maintain the original sampling rate, increase the time interval for samples, or do both. For example, increasing the interval from 1 second to 5 seconds with a 500Hz sampling rate results in 2500 timestamps per sample, which is 10 times more than the 256 timestamps per sample at 256Hz over a 1-second interval. It's important to note that contrastive learning requires maintaining a sufficient number of samples per batch. Therefore, in this scenario, you would need 10 times more GPU resources to maintain the same batch size, which may not necessarily improve the overall accuracy.

**Comment by Qu2308515296** [comment]
Hi Yihe Wang,using the TDBrain CSV dataset (99GB extracted) and your provided preprocessing code, I cannot reproduce the metrics reported in the paper. Could you advise on possible reasons? Are there specific dataset versions, splits, or preprocessing parameters I should be aware of?

**Comment by YiheWang** [comment]
Hi, can you be more specific about the reproduced results? If you get almost random results, there must be some issues with the preprocessing steps. Someone else reported the same cases, but we did not find anything wrong in the preprocessing file. In this case, leave your email, and I can send you our preprocessed version. If you get results close to our report but experience fluctuations, there are many reasons, such as differences in PyTorch or running environments (e.g., devices).

**Comment by Qu2308515296** [comment]
Please send the preprocessed data to this email：2308515296@qq.com，thanks！

**Comment by Qu2308515296** [comment]
> 你好，能否更具体地说明重现的结果？如果你得到的结果几乎是随机的，说明预处理步骤存在问题。其他人也报告了同样的情况，但我们在预处理文件中没有发现任何错误。在这种情况下，请留下你的邮箱，我可以把我们的预处理版本发给你。如果你得到的结果接近我们的报告但出现波动，原因有很多，比如PyTorch或运行环境（例如设备）的不同。

The results I get after training are indeed random.
